File: OddCode/Stickers.java
import java.util.*;

// set of stickers that spell a word
// /*google" "legolego" "leg" count_stickers(sticker_word, word)
//
// sticker word "google"
// desired word "legolego"*/
public class Stickers {
   public int countStickers(String stickerWord, String word) {
        HashMap<Character, Integer> wordCount = new HashMap<>();

        for (int i = 0; i < stickerWord.length(); i++) {
            int count = wordCount.containsKey(stickerWord.charAt(i)) ? wordCount.get(stickerWord.charAt(i)) : 0;
            wordCount.put(stickerWord.charAt(i), count + 1);
        }
        for (int i = 0; i < word.length(); i++) {
            int count = wordCount.containsKey(stickerWord.charAt(i)) ? wordCount.get(stickerWord.charAt(i)) : 0;
            if (word.charAt(i) == wordCount.get(word.charAt(i))) {
                wordCount.put(word.charAt(i), count - 1);
            }
        }
        return 5;
   } 
}

File: OddCode/Stickers.py
"""
Given a set of stickets that spell a word,
determine how many stickers you'd need to be
able to cut out enough letters to spell an
entirely different word
"""
import copy
# Reference example: stickerWord = google, word = legolego


def stickers(stickerWord, word):
    ref = {}
    valSum = []

    # Load the string
    for letter in stickerWord:
        count = 0
        if letter in ref:
            count = ref.get(letter)
        ref[letter] = count + 1

    actual = copy.deepcopy(ref)
    for letter in word:
        if letter in actual:
            count = actual.get(letter)
            actual[letter] = count - 1

    for key, value in actual.items():
        if value == 0:
            continue
        compare = ref.get(key)
        valSum.append(abs(value/compare))

    total = 0
    for number in valSum:
        total += number

    return total


print(stickers('google', 'legooooooooooooolego'))
